applications/tari_console_wallet/src/ui/components/transactions_tab.rs

```diff
@@ -283,7 +284,7 @@ impl TransactionsTab {
 }
 
 impl<B: Backend> Component<B> for TransactionsTab {
-    fn draw(&mut self, f: &mut Frame<B>, area: Rect, app_state: &AppState) {
+    fn draw(&mut self, f: &mut Frame<B>, area: Rect, app_state: &mut AppState) {
```

Comment:
`draw` was intended to only read from AppState. The idea was events change the state, and then the draw method only reads from them. That was the intention, but it's not always possible. Which part needs it to be mut?

Reply:
the state that keeps track of which item is selected in the list based on the height of the rendered list which is only available at this point.

Reply:
I think it's for the windowed items in the StatefulList?

Reply:
Correct, previously we kept the list selection state inside the component itself but for the list selection logic to work properly you would need to supply the full set of data into the render function call. To produce the windowed view into the list it had to move into the StatefulList.

Reply:
Alright, I separated the ListState offset into its own struct so it can live in the component and AppState doesn't need to be mutable.